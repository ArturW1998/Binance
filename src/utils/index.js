import { LISTS } from '../constants';

export const getTableDataListWithValues = (
  symbol,
  latestPrice,
  openPrice,
  highPrice,
  lowPrice,
  quoteVolume,
) => {
  const { TABLE_DATA_LIST } = LISTS;

  return TABLE_DATA_LIST.map(item => {
    switch (item.title) {
      case 'Pair':
        return { ...item, value: symbol, bold: true };
      case 'Latest Price':
        return { ...item, value: latestPrice };
      case 'Open':
        return { ...item, value: openPrice };
      case 'High':
        return { ...item, value: highPrice };
      case 'Low':
        return { ...item, value: lowPrice };
      case 'Volume':
        return { ...item, value: quoteVolume };
      default:
        return { ...item };
    }
  });
};

export const getTickerBySymbol = (data = []) => {
  const ticker = {};

  data.forEach(
    ({
      s: symbol,
      c: latestPrice,
      p: priceChange,
      P: priceChangePercent,
      h: highPrice,
      l: lowPrice,
      q: quoteVolume,
      o: openPrice,
    }) => {
      ticker[symbol] = {
        symbol,
        latestPrice,
        priceChange,
        priceChangePercent,
        highPrice,
        lowPrice,
        quoteVolume,
        openPrice,
      };
    },
  );

  return ticker;
};
