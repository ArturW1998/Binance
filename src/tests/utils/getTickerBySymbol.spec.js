import { getTickerBySymbol } from '../../utils';

describe('getTickerBySymbol util', () => {
  const data = [
    {
      s: 'BTCUSDT',
      c: '6626.19000000',
      h: '6842.59000000',
      l: '6573.41000000',
      q: '545581317.71130550',
      o: '6620.12000000',
    },
    {
      s: 'LRCBTC',
      c: '0.00000438',
      h: '0.00000476',
      l: '0.00000403',
      q: '86.99725405',
      o: '0.00000411',
    },
  ];

  describe('getTickerBySymbol', () => {
    it('getTickerBySymbol with data', () => {
      const {
        s: s1,
        c: c1,
        h: h1,
        l: l1,
        q: q1,
        o: o1,
        p: p1,
        P: P1,
      } = data[0];
      const {
        s: s2,
        c: c2,
        h: h2,
        l: l2,
        q: q2,
        o: o2,
        p: p2,
        P: P2,
      } = data[1];
      const expectedData = {
        [s1]: {
          symbol: s1,
          latestPrice: c1,
          priceChange: p1,
          priceChangePercent: P1,
          highPrice: h1,
          lowPrice: l1,
          quoteVolume: q1,
          openPrice: o1,
        },
        [s2]: {
          symbol: s2,
          latestPrice: c2,
          priceChange: p2,
          priceChangePercent: P2,
          highPrice: h2,
          lowPrice: l2,
          quoteVolume: q2,
          openPrice: o2,
        },
      };

      expect(getTickerBySymbol(data)).toEqual(expectedData);
    });

    it('getTickerBySymbol toMatchSnapshot', () => {
      expect(getTickerBySymbol(data)).toMatchSnapshot();
    });
  });

  describe('getTickerBySymbol initial state', () => {
    const expectedData = {};

    it('should return the initial state', () => {
      expect(getTickerBySymbol()).toEqual(expectedData);
    });

    it('returns properly', () => {
      expect(getTickerBySymbol()).toMatchSnapshot();
    });
  });
});
