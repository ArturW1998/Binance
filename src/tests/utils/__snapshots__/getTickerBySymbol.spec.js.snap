// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTickerBySymbol getTickerBySymbol toMatchSnapshot 1`] = `
Object {
  "BTCUSDT": Object {
    "highPrice": "6842.59000000",
    "latestPrice": "6626.19000000",
    "lowPrice": "6573.41000000",
    "openPrice": "6620.12000000",
    "priceChange": undefined,
    "priceChangePercent": undefined,
    "quoteVolume": "545581317.71130550",
    "symbol": "BTCUSDT",
  },
  "LRCBTC": Object {
    "highPrice": "0.00000476",
    "latestPrice": "0.00000438",
    "lowPrice": "0.00000403",
    "openPrice": "0.00000411",
    "priceChange": undefined,
    "priceChangePercent": undefined,
    "quoteVolume": "86.99725405",
    "symbol": "LRCBTC",
  },
}
`;

exports[`getTickerBySymbol initial state returns properly 1`] = `Object {}`;
